istioctl/cmd/gkeutils.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"istio.io/pkg/log"
+
+	container "cloud.google.com/go/container/apiv1"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
```

Comment:
I am nervous about putting Google GKE specific code into _istioctl_.  It opens the door to putting in IBM or AWS code.  Is there a way to factor this so the GKE code does not need to be included with _istioctl_?